test/levels/Switch.test.js

```diff
@@ -0,0 +1,83 @@
+const Switch = artifacts.require('./levels/Switch.sol')
+const SwitchFactory = artifacts.require('./levels/SwitchFactory.sol')
+
+const Ethernaut = artifacts.require('./Ethernaut.sol')
+
+import * as utils from '../utils/TestUtils'
+import expectThrow from 'zeppelin-solidity/test/helpers/expectThrow'
+import toPromise from 'zeppelin-solidity/test/helpers/toPromise'
+
+contract('Switch', function(accounts) {
+
+  let ethernaut
+  let level
+  let instance
+  let player = accounts[0]
+
+  before(async function() {
+    ethernaut = await Ethernaut.new();
+    level = await SwitchFactory.new()
+    await ethernaut.registerLevel(level.address)
+    instance = await utils.createLevelInstance(
+      ethernaut, level.address, player, Switch,
+      {from: player}
+    )
+  });
+
+  describe('instance', function() {
+
+    it('should not be immediately solvable', async function() {
+
+      // make sure the factory fails
+      const ethCompleted = await utils.submitLevelInstance(
+        ethernaut,
+        level.address,
+        instance.address,
+        player
+      )
+      assert.equal(ethCompleted, false);
+    });
+
+    it('it should allow the user to solve the level', async function() {
+
+      // build the necessary function selectors
+      let flipSwitchSelector = web3.sha3("flipSwitch(bytes)").substring(2,10)
```

Comment:
web.sha3 function is not found, should be replaced with web3.utils.soliditySha3

```suggestion
      let flipSwitchSelector = web3.utils.keccak256("flipSwitch(bytes)").substring(2,10)
      let onFunctionSelector = web3.utils.keccak256("turnSwitchOn()").substring(2,10)
      let offFunctionSelector = web3.utils.keccak256("turnSwitchOff()").substring(2,10)
```